autogen/main/cluster.tf.tmpl

```diff
@@ -89,10 +89,18 @@ resource "google_container_cluster" "primary" {
   }
   monitoring_service = local.cluster_telemetry_type_is_set || local.logmon_config_is_set ? null : var.monitoring_service
   dynamic "monitoring_config" {
-    for_each = length(var.monitoring_enabled_components) > 0 ? [1] : []
+    for_each = length(var.monitoring_enabled_components) > 0 || var.monitoring_enable_managed_prometheus ? [1] : []
 
     content {
-      enable_components = var.monitoring_enabled_components
+      enable_components = length(var.monitoring_enabled_components) > 0 ? var.monitoring_enabled_components : null
+
+      dynamic "managed_prometheus" {
+        for_each = var.monitoring_enable_managed_prometheus ? [1] : []
+
+        content {
+          enabled = var.monitoring_enable_managed_prometheus
+        }
+      }
```

Comment:
You can also enable `managed_prometheus` if `autopilot_cluster` is `true` (but Autopilot cannot enable `enable_components`).

Reply:
@ykzts Can you open an issue so we can track this for autopilot support?